tests/console/verify_systemd_timesync.pm

```diff
@@ -0,0 +1,48 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Verify that systemd timer is used for time synchronization.
+# Maintainer: Sofia Syrianidou <ssyrianidou@suse.com>
+
+use strict;
+use warnings;
+use base 'y2_installbase';
+use testapi;
+use scheduler 'get_test_suite_data';
+use Test::Assert ':all';
+
+sub run {
+    my $test_data = get_test_suite_data();
+    select_console 'root-console';
+    record_info("Check yast-timesync.timer", "Check if file /etc/systemd/system/yast-timesync.timer exists and has expected configuration.");
+    assert_script_run('ls /etc/systemd/system/yast-timesync.timer');
+    my $conf_file = script_output("cat /etc/systemd/system/yast-timesync.timer");
+    $conf_file =~ /OnUnitActiveSec=(?<interval>(\S+))min/;
+    my $expected_interval = $test_data->{profile}->{'ntp-client'}->{ntp_sync};
+    assert_equals($expected_interval, $+{interval}, "The interval in yast-timesync configuration file isnot the expected one.");
+    $conf_file =~ /OnBootSec=(?<first_sync>(\S+))min/;
+    sleep($+{first_sync} * 60);
+    record_info("Check sync", "Check if the use of systemd timers can be spotted in the journal logs as expected.");
+    assert_script_run("journalctl -u yast-timesync | grep \"Started One time sync configured by YaST\"");
+    record_info("Deviation fix", "Check if \"One time synchronization\" will fix time deviation as expected.");
+    my $last_sync_time = script_output("journalctl -u yast-timesync --since \"$expected_interval minutes ago\"| grep \"Starting One time\" | awk '{print \$3}'");
+    assert_script_run("date");
+    assert_script_run('date $(date +%m%d%H%M%Y.%S -d "1 minute ago")');
+    # Calculating time of next "One time timesync" minus 5 seconds for safety.
+    my $next_sync_time = (split(":", $last_sync_time))[0] . ":" . ((split(":", $last_sync_time))[1] + $expected_interval) . ":" . (split(":", $last_sync_time) - 5)[2];
```

Comment:
Actually, that was not very thoughtful of me as in case of next_sync_time would be for example 14:01:46 this would fail miserably, it would give 13:61:41 . I used the DateTime library, which is available in OpenQA workers, but now, with current version of script I don't need this.